conda_index/index/sqlitecache.py

```diff
@@ -83,21 +87,33 @@ def __get__(self, inst, objtype=None) -> Any:
 class CondaIndexCache:
     upstream_stage = "fs"
 
-    def __init__(self, channel_root, subdir):
+    def __init__(
+        self,
+        channel_root: Path | str,
+        subdir: str,
+        *,
+        fs: MinimalFS | None = None,
+        channel_url: str | None = None,
+    ):
         """
         channel_root: directory containing platform subdir's, e.g. /clones/conda-forge
         subdir: platform subdir, e.g. 'linux-64'
+        fs: MinimalFS (designed to wrap fsspec.spec.AbstractFileSystem); optional.
+        channel_url: base url if fs is used; optional.
         """
-        self.channel_root = channel_root
+
         self.subdir = subdir
+        self.channel_root = Path(channel_root)
+        self.subdir_path = Path(channel_root, subdir)
+        self.cache_dir = Path(channel_root, subdir, ".cache")
+        self.db_filename = self.cache_dir / "cache.db"
+        self.cache_is_brand_new = not Path(self.db_filename).exists()
 
-        self.subdir_path = os.path.join(channel_root, subdir)
-        self.cache_dir = os.path.join(self.subdir_path, ".cache")
-        self.db_filename = os.path.join(self.cache_dir, "cache.db")
-        self.cache_is_brand_new = not os.path.exists(self.db_filename)
+        self.fs = fs or MinimalFS()
+        self.channel_url = channel_url or str(channel_root)
```

Comment:
Should we check here too that both are defined/undefined before defaulting?

Reply:
Maybe? We only expect this to be instantiated (correctly) by ChannelIndex()